runtime/jvmMain/src/kotlinx/serialization/internal/Platform.kt

```diff
@@ -43,18 +43,31 @@ internal actual fun <T : Any> KClass<T>.constructSerializerForGivenTypeArgs(vara
     // Check whether it's serializable object
     findObjectSerializer(jClass)?.let { return it }
     // Search for default serializer if no serializer is defined in companion object.
-    return try {
+    val default = try {
         jClass.declaredClasses.singleOrNull { it.simpleName == ("\$serializer") }
             ?.getField("INSTANCE")?.get(null) as? KSerializer<T>
     } catch (e: NoSuchFieldException) {
         null
     }
+
+    if (default != null) return default
+    // Fallback: enum without @Serializable annotation as last resort
+    return createEnumSerializer()
+}
+
+@Suppress("UNCHECKED_CAST")
+private fun <T : Any> KClass<T>.createEnumSerializer(): KSerializer<T>? {
+    if (!Enum::class.java.isAssignableFrom(java)) return null
```

Comment:
```suggestion
    if (!Enum::class.java.isAssignableFrom(this.java)) return null
```

slightly more readable